### ISSUE TITLE
Error on make: src/asm/fstack.c:208:10: error: 'ENOENT' undeclared (first use in this function)

### ISSUE THREAD

**Issue by arteze** [gbdev/rgbds#906]
I have an error when executing `make`:

```sh
/rgbds-master
make
src/asm/fstack.c: In function 'fstk_FindFile':
src/asm/fstack.c:208:10: error: 'ENOENT' undeclared (first use in this function)
  208 |  errno = ENOENT;
      |          ^~~~~~
src/asm/fstack.c:208:10: note: each undeclared identifier is reported only once for each function it appears in
make: *** [Makefile:148: src/asm/fstack.o] Error 1
```

¿It is possible to solve it?

**Comment by Rangi42** [comment]
`ENOENT` should have been defined by `#include <errno.h>` at the top of `src/asm/fstack.c`. What is your OS and compiler?

**Comment by arteze** [comment]
@Rangi42 I have puppylinux... I see that it is including `#include <errno.h>` in line `/src/asm/fstack.c:11`.

Then I imagine that the problem is inside the `errno.h`.

**Comment by arteze** [comment]
I already solved it, I added this:

```c
#include <asm-generic/errno-base.h>
```

Now I have the mistake that I need `pow`:

```sh
/rgbds-master
make --trace
Makefile:121: update target 'rgbgfx' due to: src/gfx/gb.o src/gfx/main.o src/gfx/makepng.o src/extern/err.o src/extern/getopt.o
cc  -Wall -DBUILD_VERSION_STRING=\"`git describe --tags --dirty --always 2>/dev/null`\" `pkg-config --libs-only-L libpng` -o rgbgfx src/gfx/gb.o src/gfx/main.o src/gfx/makepng.o src/extern/err.o src/extern/getopt.o -O3 -flto -DNDEBUG -Wall -std=gnu11 -I include -D_POSIX_C_SOURCE=200809L -D_ISOC11_SOURCE src/version.c `pkg-config --libs-only-l libpng`
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libpng16.a(png.o): in function `png_build_16bit_table':
(.text+0x5fe): undefined reference to `pow'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libpng16.a(png.o): in function `png_build_8bit_table':
(.text+0xdc8): undefined reference to `pow'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libpng16.a(png.o): in function `png_colorspace_set_endpoints':
(.text+0x48a1): undefined reference to `floor'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libpng16.a(png.o): in function `png_gamma_8bit_correct':
(.text+0x4c6d): undefined reference to `pow'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libpng16.a(png.o): in function `png_gamma_16bit_correct':
(.text+0x4d2d): undefined reference to `pow'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libpng16.a(png.o): in function `png_gamma_correct':
(.text+0x4e11): undefined reference to `pow'
/usr/bin/ld: (.text+0x4ea1): undefined reference to `pow'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libpng16.a(png.o): in function `png_build_gamma_table':
(.text+0x53c7): undefined reference to `pow'
collect2: error: ld returned 1 exit status
make: *** [Makefile:121: rgbgfx] Error 1
```

**Comment by arteze** [comment]
@Rangi42 I forgot to mention the version of my compiler:

`gcc -v
Reading specs from /usr/lib64/gcc/x86_64-slackware-linux/10.3.0/specs
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-slackware-linux/10.3.0/lto-wrapper
Target: x86_64-slackware-linux
Configured with: ../configure --prefix=/usr --libdir=/usr/lib64 --mandir=/usr/man --infodir=/usr/info --enable-shared --enable-bootstrap --enable-languages=ada,brig,c,c++,d,fortran,go,lto,objc,obj-c++ --enable-threads=posix --enable-checking=release --enable-objc-gc --with-system-zlib --enable-libstdcxx-dual-abi --with-default-libstdcxx-abi=new --disable-libstdcxx-pch --disable-libunwind-exceptions --enable-__cxa_atexit --disable-libssp --enable-gnu-unique-object --enable-plugin --enable-lto --disable-install-libiberty --disable-werror --with-gnu-ld --with-isl --verbose --with-arch-directory=amd64 --disable-gtktest --enable-clocale=gnu --disable-multilib --target=x86_64-slackware-linux --build=x86_64-slackware-linux --host=x86_64-slackware-linux
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 10.3.0 (GCC) `

**Comment by ISSOtm** [comment]
You shouldn't have to include extra headers; if `<errno.h>` isn't providing `ENOENT`, there's a problem. Maybe a feature test macro? The specific header you're including looks really dubious, too.

As for `pow`, it's being required by libpng, so `pkg-config --libs-only-l` should produce `-lm` if necessary. Just to be safe, what is `pkg-config --libs` producing on your machine?

**Comment by arteze** [comment]
I was able to solve it, I removed this part of the command `gcc`:

```sh
`pkg-config --libs-only-L libpng`
```

`cc  -Wall -DBUILD_VERSION_STRING=\"'git describe --tags --dirty --always 2>/dev/null'\" 'pkg-config --libs-only-L libpng' -o rgbgfx src/gfx/gb.o src/gfx/main.o src/gfx/makepng.o src/extern/err.o src/extern/getopt.o -O3 -flto -DNDEBUG -Wall -std=gnu11 -I include -D_POSIX_C_SOURCE=200809L -D_ISOC11_SOURCE src/version.c 'pkg-config --libs-only-l libpng`

Being that way:

```sh
cc  -Wall -DBUILD_VERSION_STRING=\"`git describe --tags --dirty --always 2>/dev/null`\" -o rgbgfx src/gfx/gb.o src/gfx/main.o src/gfx/makepng.o src/extern/err.o src/extern/getopt.o -O3 -flto -DNDEBUG -Wall -std=gnu11 -I include -D_POSIX_C_SOURCE=200809L -D_ISOC11_SOURCE src/version.c `pkg-config --libs-only-l libpng` -lpng
```

You can close the issue.

**Comment by ISSOtm** [comment]
If you want, but these fixes are not correct. Whatever works for you, I guess.

**Comment by arteze** [comment]
@ISSOtm The `pkgconfig` It produces the following output:

```sh
/rgbds-master
pkg-config --libs
Must specify package names on the command line
```

**Comment by arteze** [comment]
I see that it works if I do `pkg-config libpng`.

**Comment by arteze** [comment]
Now, the correct command returns me the following:

```sh
/rgbds-master
pkg-config --libs-only-L libpng
-L/usr/lib/x86_64-linux-gnu
```

If I do the following, do not compile:

```sh
cc  -Wall -DBUILD_VERSION_STRING=\"`git describe --tags --dirty --always 2>/dev/null`\" -o rgbgfx src/gfx/gb.o src/gfx/main.o src/gfx/makepng.o src/extern/err.o src/extern/getopt.o -O3 -flto -DNDEBUG -Wall -std=gnu11 -I include -D_POSIX_C_SOURCE=200809L -D_ISOC11_SOURCE src/version.c `pkg-config --libs-only-l libpng` -L/usr/lib/x86_64-linux-gnu
```

**Comment by ISSOtm** [comment]
It doesn't make sense that *adding* lib search paths causes the build to fail... :raised_eyebrow:

**Comment by arteze** [comment]
Actually it does make sense, simply those inclusions can be damaged.

Well, just just spoiling the whole system and I had to format the computer. Now I should reinstall all the programs, is that Puppylinux has many bugs, but I'm already used to it.

When I compiled it without including the patch of the bookstores, also that the program worked very well, it worked correctly, I was able to compile a video game without problems.
